### ISSUE TITLE
`toValue` consider function given as a factory.

### ISSUE THREAD

**Issue by liplum** [owja/ioc#63]
**Describe the bug**
The Container.bind(...).toValue(...) consider the a given function as a fatory.

**To Reproduce**
```js
import { Container, token } from "@owja/ioc"
const TYPE = {
  myType: token("MyType")
}
const container = new Container()
container.bind(TYPE.myType).toValue((text) => {
  if (text === undefined) console.log("You cannot give undefined.")
  else console.log(text)
})
const myFunc = container.get(TYPE.myType)
myFunc("hello, world!")
```
the console output is 
```
You cannot give undefined.
file:///test.js:11
myFunc("hello, world!")
^

TypeError: myFunc is not a function
```
**Expected behavior**
I want to bind a Getter function, such as (options) => create(options) with `toValue`, however, the container considered it as a factory and tried to call it and retrieved the return value.

I don't think `toValue()` should detect its type under the hood, which made me confused.
Instead, `toFactory()` is more suitable for this.

**Environment (please complete the following information):**
 - typescript^4.9.5
 - Node v16.18.0
 - ^2.0.0-alpha.6

**Additional context**
The workaround is to wrap the function with a new function.
```js
container.bind(TYPE.myType).toValue(() => (text) => {
  if (text === undefined) console.log("You cannot give undefined.")
  else console.log(text)
})
```
But it makes the typing sucks in typescript.
Because the token should be `token<()=>(text:string)=>void>("MyType")`, then the return value from `container.get(...)` is typed by `()=>(text:string)=>void` but `(text:string)=>void` is actually expected.

**Comment by liplum** [comment]
After I checked the source code, I found the factory checking in `Container.get`.
```ts
get<T, U extends Array<unknown> = never>(
    token: Token<T, U> | MaybeToken<T>,
    tags: symbol[] | symbol = [],
    target?: unknown,
    injectedArgs: Array<unknown> = [],
): T {
    const item = <Item<T> | undefined>this._registry.get(getType(token));
    if (item === undefined || item.injected === undefined) throw `nothing bound to ${stringifyToken(token)}`;
    const value = isFactory(item.injected)
        ? !item.singleton
            ? item.injected(...injectedArgs)
            : (item.cache = item.cache || item.injected())
        : item.injected;
    const tagsArr = valueOrArrayToArray(tags);
    if (tagsArr.indexOf(NOPLUGINS) === -1)
        item.plugins.concat(this._plugins).forEach((plugin) => {
            plugin(value, target, tagsArr, token, this);
        });
    return value;
}
```
It's better to separate FactoryItem from Item.
I would make a pull request for this.

**Comment by hbroer** [comment]
@liplum i published a new version (2.0.0-alpha.7). That should fix the bug.

**Comment by liplum** [comment]
Yes, it works.
